### PR TITLE
F#33 combining projectiles animations

### PR DESCRIPTION
**Pull request by Dejin1**

### CODE REVIEW

File: Assets/Scripts/Enemy/Enemy.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@ public class Enemy : MonoBehaviour {
     // Use this for initialization
     public void TakeDamage (int damage)
     {
+        //Debug.Log("damage called");
```
Comment (massieb) [inline Assets/Scripts/Enemy/Enemy.cs]: Remove commented code

File: Assets/Scripts/Firing/Weapon.cs [hunk 2]
```diff
@@ -9,16 +9,15 @@ public class Weapon : MonoBehaviour
     public GameObject bulletPrefab;
     private float lastShotTime;
     public float shotDelay = 2;
-    public static string shootProjectile;
-    public string setStringForProjectile = "Fire1";
+    
+    
 
     //*All code commented out is in effect until assets have been aquired.*
```
Comment (massieb) [inline Assets/Scripts/Firing/Weapon.cs]: This line is not required anymore. Commented code is in effect

File: Assets/Scripts/GameControlScript.cs [hunk 3]
```diff
@@ -60,6 +60,32 @@ void LateUpdate()
...
+                JinPlayerMovement.canMove = false;
+                gameOver.gameObject.SetActive(true);
+                break;
+        }
+
+        switch (Player1.GetComponent<Health>()._HP)
+        {
+            case 3:
+                //Full health
+
```
Comment (massieb) [inline Assets/Scripts/GameControlScript.cs]: Unnecessary white-space in all cases

File: Assets/Scripts/Player/CharacterController2D.cs [hunk 4]
```diff
@@ -147,29 +146,24 @@ public void Move(float move, bool crouch, bool jump, bool getDownFromPlatform)
...
             // If the input is moving the player right and the player is facing left...
             if (move > 0 && !m_FacingRight)
             {
-                //Debug.Log("FacingRight");
-                Flip();
-
+                Flip(true);
             }
             // Otherwise if the input is moving the player left and the player is facing right...
             else if (move < 0 && m_FacingRight)
```
Comment (massieb) [inline Assets/Scripts/Player/CharacterController2D.cs]: You can combine this with the first if
`if ((move > 0 && !m_FacingRight) || (move < 0 && m_FacingRight))`

File: Assets/Scripts/Player/CharacterController2D.cs [hunk 5]
```diff
@@ -196,28 +190,28 @@ public void CheckIfLR()
...
-            transform.localScale = theScale;
+            Flip(false);
         }
     }
 
-    private void Flip()
+    /// <summary>
+    /// This flips the sprite taking a bool
+    /// </summary>
+    /// <param name="flipSprite"></param>
```
Comment (massieb) [inline Assets/Scripts/Player/CharacterController2D.cs]: missing comments of the param (needed when you add it to the comments) you could also remove it when its clear what the param is/does.

File: Assets/Scripts/Player/CharacterController2D.cs [hunk 6]
```diff
@@ -196,28 +190,28 @@ public void CheckIfLR()
...
+    private void Flip(bool flipSprite)
```
Comment (massieb) [inline Assets/Scripts/Player/CharacterController2D.cs]: Method flip does not always flip according to the boolean. Rename the method to give a better indication of what it does

File: Assets/Scripts/Player/JinPlayerMovement.cs [hunk 7]
```diff
@@ -18,8 +18,8 @@ public class JinPlayerMovement : MonoBehaviour
     public string stringFloat;
     public string stringInteract;
 
-    public static string shootProjectile = "Fire1";
-
+    public static string shootProjectile;
+    public string setStringForProjectile = "";
```
Comment (massieb) [inline Assets/Scripts/Player/JinPlayerMovement.cs]: Variable has a method-like name, very confusing

File: Assets/Scripts/Player/JinPlayerMovement.cs [hunk 8]
```diff
@@ -196,11 +191,12 @@ void Update()
 
             if (Input.GetButtonDown(shootProjectile))
             {
-                //Todo: Once combined with animation branch this code can be uncommented
-                //if (CharacterAnimator.GetBool("isCrouching") == false)
-                //{
-                gameObject.GetComponent<Weapon>().Shoot();
-                //}
+                
+                if (animator.GetBool("isCrouching") == false)
```
Comment (massieb) [inline Assets/Scripts/Player/JinPlayerMovement.cs]: extract hardcoded string (already used twice in this file)

Comment (massieb) [inline Assets/Scripts/Player/JinPlayerMovement.cs]: Change to convenience usage of booleans (equal check is not required)
`if (!animator.GetBool("isCrouching"))`

File: Assets/Scripts/Player/JinPlayerMovement.cs [hunk 9]
```diff
@@ -211,6 +207,7 @@ void Update()
     public void OnCrouching(bool isCrouching)
```
Comment (massieb) [inline Assets/Scripts/Player/JinPlayerMovement.cs]: Why does this method requires a boolean? The method already indicates that the player is crouching

File: Assets/Scripts/Enemy/Enemy.cs (hunk 1 above)
Comment (Dejin1) [reply Assets/Scripts/Enemy/Enemy.cs]: Fixed in new commit

File: Assets/Scripts/Firing/Weapon.cs (hunk 2 above)
Comment (Dejin1) [reply Assets/Scripts/Firing/Weapon.cs]: I am sorry for not catching that lol

File: Assets/Scripts/GameControlScript.cs (hunk 3 above)
Comment (Dejin1) [reply Assets/Scripts/GameControlScript.cs]: These were always here but i shall remove them and add to do's next to the comments

Comment (Dejin1) [reply Assets/Scripts/GameControlScript.cs]: Fixed in new commit

File: Assets/Scripts/Player/CharacterController2D.cs (hunk 4 above)
Comment (Dejin1) [reply Assets/Scripts/Player/CharacterController2D.cs]: Fixed in new commit I guess i need to work on my linear thinking when it comes to if's lol

File: Assets/Scripts/Player/CharacterController2D.cs (hunk 6 above)
Comment (Dejin1) [reply Assets/Scripts/Player/CharacterController2D.cs]: This method always flips weather or not the bool is true this code:

Flips the character weather or not the parameter is true or false

What would be a better name then?

File: Assets/Scripts/Player/JinPlayerMovement.cs (hunk 7 above)
Comment (Dejin1) [reply Assets/Scripts/Player/JinPlayerMovement.cs]: changed from setstring to inputString in new commit

File: Assets/Scripts/Player/JinPlayerMovement.cs (hunk 8 above)
Comment (Dejin1) [reply Assets/Scripts/Player/JinPlayerMovement.cs]: fixed in new commit thanks marcel :)

Comment (Dejin1) [reply Assets/Scripts/Player/JinPlayerMovement.cs]: fixed in new commit

File: Assets/Scripts/Player/JinPlayerMovement.cs (hunk 9 above)
Comment (Dejin1) [reply Assets/Scripts/Player/JinPlayerMovement.cs]: This is checking for if crouching fixed name so its less confusing fixed in new commit

File: Assets/Scripts/Player/CharacterController2D.cs (hunk 5 above)
Comment (Dejin1) [reply Assets/Scripts/Player/CharacterController2D.cs]: fixed in new commit

**Pull request merged**